### ISSUE TITLE
Build fails with GCC 10.2 and PyPy 7.3.3

### ISSUE THREAD

**Issue by sjuxax** [closeio/ciso8601#101]
I have versions:

```
$ pypy3 --version
Python 3.7.9 (7e6e2bb30ac5fbdbd443619cae28c51d5c162a02, Nov 24 2020, 10:03:59)
[PyPy 7.3.3-beta0 with GCC 10.2.0]

$ gcc --version
gcc (GCC) 10.2.0
```

When I run `CFLAGS="$CFLAGS -fcommon" pypy3 -m pip install git+https://github.com/closeio/ciso8601.git`, I get:

```
Defaulting to user installation because normal site-packages is not writeable
Collecting git+https://github.com/closeio/ciso8601.git
  Cloning https://github.com/closeio/ciso8601.git to /tmp/pip-req-build-zank7ym7
  Running command git clone -q https://github.com/closeio/ciso8601.git /tmp/pip-req-build-zank7ym7
Building wheels for collected packages: ciso8601
  Building wheel for ciso8601 (setup.py): started
  Building wheel for ciso8601 (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/pypy3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-zank7ym7/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-zank7ym7/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-k0gy_icr
       cwd: /tmp/pip-req-build-zank7ym7/
  Complete output (27 lines):
  running bdist_wheel
  running build
  running build_py
  package init file 'ciso8601/__init__.py' not found (or not a regular file)
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/ciso8601
  copying ciso8601/__init__.pyi -> build/lib.linux-x86_64-3.7/ciso8601
  copying ciso8601/py.typed -> build/lib.linux-x86_64-3.7/ciso8601
  running build_ext
  building 'ciso8601' extension
  creating build/temp.linux-x86_64-3.7
  gcc -pthread -DNDEBUG -O2 -O2 -march=broadwell -mmmx -mno-3dnow -msse -msse2 -msse3 -mssse3 -mno-sse4a -mcx16 -msahf -mmovbe -maes -mno-sha -mpclmul -mpopcnt -mabm -mno-lwp -mfma -mno-fma4 -mno-xop -mbmi -mno-sgx -mbmi2 -mno-pconfig -mno-wbnoinvd -mno-tbm -mavx -mavx2 -msse4.2 -msse4.1 -mlzcnt -mrtm -mhle -mrdrnd -mf16c -mfsgsbase -mrdseed -mprfchw -madx -mfxsr -mxsave -mxsaveopt -mno-avx512f -mno-avx512er -mno-avx512cd -mno-avx512pf -mno-prefetchwt1 -mno-clflushopt -mno-xsavec -mno-xsaves -mno-avx512dq -mno-avx512bw -mno-avx512vl -mno-avx512ifma -mno-avx512vbmi -mno-avx5124fmaps -mno-avx5124vnniw -mno-clwb -mno-mwaitx -mno-clzero -mno-pku -mno-rdpid -mno-gfni -mno-shstk -mno-avx512vbmi2 -mno-avx512vnni -mno-vaes -mno-vpclmulqdq -mno-avx512bitalg -mno-movdiri -mno-movdir64b -mno-waitpkg -mno-cldemote -mno-ptwrite -mtune=broadwell -pipe -fstack-protector-explicit -fno-plt -g -gdwarf -gz -ggdb -Wno-error=deprecated-declarations -fcommon -fPIC -DCISO8601_VERSION=2.1.3 -I/opt/pypy3/include -c module.c -o build/temp.linux-x86_64-3.7/module.o
  module.c: In function ‘_parse’:
  module.c:446:30: warning: implicit declaration of function ‘PyTimeZone_FromOffset’ [-Wimplicit-function-declaration]
    446 |                     tzinfo = PyTimeZone_FromOffset(delta);
        |                              ^~~~~~~~~~~~~~~~~~~~~
  module.c:446:28: warning: assignment to ‘PyObject *’ {aka ‘struct _object *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
    446 |                     tzinfo = PyTimeZone_FromOffset(delta);
        |                            ^
  module.c: In function ‘PyInit_ciso8601’:
  module.c:562:11: error: ‘PyDateTime_TimeZone_UTC’ undeclared (first use in this function); did you mean ‘PyDateTime_Time’?
    562 |     utc = PyDateTime_TimeZone_UTC;
        |           ^~~~~~~~~~~~~~~~~~~~~~~
        |           PyDateTime_Time
  module.c:562:11: note: each undeclared identifier is reported only once for each function it appears in
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for ciso8601
  Running setup.py clean for ciso8601
Failed to build ciso8601
Installing collected packages: ciso8601
    Running setup.py install for ciso8601: started
    Running setup.py install for ciso8601: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/pypy3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-zank7ym7/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-zank7ym7/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-bxkrj8i3/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/jeff/.local/include/python3.7/ciso8601
         cwd: /tmp/pip-req-build-zank7ym7/
    Complete output (27 lines):
    running install
    running build
    running build_py
    package init file 'ciso8601/__init__.py' not found (or not a regular file)
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/ciso8601
    copying ciso8601/__init__.pyi -> build/lib.linux-x86_64-3.7/ciso8601
    copying ciso8601/py.typed -> build/lib.linux-x86_64-3.7/ciso8601
    running build_ext
    building 'ciso8601' extension
    creating build/temp.linux-x86_64-3.7
    gcc -pthread -DNDEBUG -O2 -O2 -march=broadwell -mmmx -mno-3dnow -msse -msse2 -msse3 -mssse3 -mno-sse4a -mcx16 -msahf -mmovbe -maes -mno-sha -mpclmul -mpopcnt -mabm -mno-lwp -mfma -mno-fma4 -mno-xop -mbmi -mno-sgx -mbmi2 -mno-pconfig -mno-wbnoinvd -mno-tbm -mavx -mavx2 -msse4.2 -msse4.1 -mlzcnt -mrtm -mhle -mrdrnd -mf16c -mfsgsbase -mrdseed -mprfchw -madx -mfxsr -mxsave -mxsaveopt -mno-avx512f -mno-avx512er -mno-avx512cd -mno-avx512pf -mno-prefetchwt1 -mno-clflushopt -mno-xsavec -mno-xsaves -mno-avx512dq -mno-avx512bw -mno-avx512vl -mno-avx512ifma -mno-avx512vbmi -mno-avx5124fmaps -mno-avx5124vnniw -mno-clwb -mno-mwaitx -mno-clzero -mno-pku -mno-rdpid -mno-gfni -mno-shstk -mno-avx512vbmi2 -mno-avx512vnni -mno-vaes -mno-vpclmulqdq -mno-avx512bitalg -mno-movdiri -mno-movdir64b -mno-waitpkg -mno-cldemote -mno-ptwrite -mtune=broadwell -pipe -fstack-protector-explicit -fno-plt -g -gdwarf -gz -ggdb -Wno-error=deprecated-declarations -fcommon -fPIC -DCISO8601_VERSION=2.1.3 -I/opt/pypy3/include -c module.c -o build/temp.linux-x86_64-3.7/module.o
    module.c: In function ‘_parse’:
    module.c:446:30: warning: implicit declaration of function ‘PyTimeZone_FromOffset’ [-Wimplicit-function-declaration]
      446 |                     tzinfo = PyTimeZone_FromOffset(delta);
          |                              ^~~~~~~~~~~~~~~~~~~~~
    module.c:446:28: warning: assignment to ‘PyObject *’ {aka ‘struct _object *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
      446 |                     tzinfo = PyTimeZone_FromOffset(delta);
          |                            ^
    module.c: In function ‘PyInit_ciso8601’:
    module.c:562:11: error: ‘PyDateTime_TimeZone_UTC’ undeclared (first use in this function); did you mean ‘PyDateTime_Time’?
      562 |     utc = PyDateTime_TimeZone_UTC;
          |           ^~~~~~~~~~~~~~~~~~~~~~~
          |           PyDateTime_Time
    module.c:562:11: note: each undeclared identifier is reported only once for each function it appears in
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/pypy3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-zank7ym7/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-zank7ym7/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-bxkrj8i3/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/jeff/.local/include/python3.7/ciso8601 Check the logs for full command output.
```

Primary issue appears to be missing `PyDateTime_TimeZone_UTC`, which is new in Python 3.7 but this version of PyPy claims to implement Python 3.7.

GCC 10 causes these issues sometimes since it switched to the default of `-fno-common`, but that doesn't seem to matter here since adding `-fcommon` to the CFLAGS doesn't change anything.

**Comment by movermeyer** [comment]
Hey @sjuxax,

ciso8601 currently does not support PyPy. 

If you investigate further and find out that it would be simple to add such support, then feel free to open a PR. 
But it may also turn out to be simply "out of scope".

**Comment by movermeyer** [comment]
I expect that this will be fixed implicitly by #58, given that it would make `ciso8601` no longer use `PyDateTime_TimeZone_UTC`. 🎉

**Comment by movermeyer** [comment]
Related issue in upstream Pypy: https://foss.heptapod.net/pypy/pypy/-/issues/3320
